src/sock.c

```diff
@@ -234,6 +234,25 @@ struct dnsquery_resourcerecord
 	struct dnsquery_srvrdata rdata;
 };
 
+void netbuf_add_32bitnum(unsigned char *, int, int *, unsigned int);
```

Comment:
As I see, all these declarations are placed just above their definitions. Since these functions don't call each other the declarations are not necessary and only increase LOC and complexity. Please, remove them.